dlt/destinations/synapse/sql_client.py

```diff
@@ -0,0 +1,222 @@
+        self._conn: pyodbc.Connection = None
+        self.credentials = credentials
+
+    def open_connection(self) -> pyodbc.Connection:
+        self._conn = pyodbc.connect(
+            self.credentials.to_odbc_dsn(),
+            timeout=self.credentials.connect_timeout,
+        )
+        self._conn.add_output_converter(-155, handle_datetimeoffset)
+        self._conn.autocommit = True
+        print("Conn string: ", self.credentials.to_odbc_dsn())
+        return self._conn
+
+    @raise_open_connection_error
+    def close_connection(self) -> None:
+        if self._conn:
+            self._conn.close()
+            self._conn = None
+
+    @contextmanager
+    def begin_transaction(self) -> Iterator[DBTransaction]:
+        try:
+            self._conn.autocommit = False
+            yield self
+            self.commit_transaction()
+        except Exception:
+            self.rollback_transaction()
+            raise
+
+    @raise_database_error
+    def commit_transaction(self) -> None:
+        self._conn.commit()
+        self._conn.autocommit = True
+
+    @raise_database_error
+    def rollback_transaction(self) -> None:
+        self._conn.rollback()
+        self._conn.autocommit = True
+
+    @property
+    def native_connection(self) -> pyodbc.Connection:
+        return self._conn
+
+    def drop_dataset(self) -> None:
+        # Save the current autocommit state
+        # If a transaction is active, commit it
+        if not self._conn.autocommit:
+            self._conn.commit()
+
+        # Save the current autocommit state
+        prev_autocommit = self._conn.autocommit
+        # Set autocommit to True for DDL operations
+        self._conn.autocommit = True
+        try:
+            rows = self.execute_sql(
+                "SELECT table_name FROM information_schema.views WHERE table_schema = %s;", self.dataset_name
+            )
+            view_names = [row[0] for row in rows]
+            self._drop_views(*view_names)
+            rows = self.execute_sql(
+                "SELECT table_name FROM information_schema.tables WHERE table_schema = %s;", self.dataset_name
+            )
+            table_names = [row[0] for row in rows]
+            self.drop_tables(*table_names)
+
+            self.execute_sql("DROP SCHEMA %s;" % self.fully_qualified_dataset_name())
+        finally:
+            # Reset autocommit to its previous state
+            self._conn.autocommit = prev_autocommit
+
+    def table_exists(self, table_name: str) -> bool:
+        query = """
+        SELECT COUNT(*) 
+        FROM INFORMATION_SCHEMA.TABLES 
+        WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ?
+        """
+        result = self.execute_sql(query, self.dataset_name, table_name)
+        return result[0][0] > 0        
+
+    def drop_tables(self, *tables: str) -> None:
+        if not tables:
+            return
+        # If a transaction is active, commit it
+        if not self._conn.autocommit:
```

Comment:
i don't think you need this kind of code here. In which case is a transaction avtive when you drop the dataset?